### PR TITLE
Optimize FastAIModel for reduced peak memory during inference 

### PR DESCRIPTION
**Pull request by dthaler**
Implements memory and cleanup optimizations for FastAIModel inference.

## Changes Made

- **Add batch_size and use_gpu parameters**: Constructor now accepts `batch_size` (default 32) and `use_gpu` (default False) parameters
- **Device placement in init**: Model is moved to GPU/CPU once during initialization instead of on every predict() call
- **try/finally cleanup**: Ensures temp WAV files are cleaned up even on exceptions
- **Cleanup logging**: Temp directory cleanup failures are now logged instead of silently ignored
- **Explicit memory release**: Deletes large fastai objects (`test`, `testdb`) after processing and calls `gc.collect()` + `torch.cuda.empty_cache()`
- **Other fixes**: `pd.concat()` instead of deprecated `append()`, fixed typos

## Note on Batched Inference

The original goal of using `get_preds()` for batched inference couldn't be implemented because FastAI 1.x's DataLoader ordering doesn't match `test_data_folder.ls()` ordering, causing path-prediction alignment issues. The `batch_size` parameter is used for DataLoader creation but predictions are made per-item using fastai's `predict()` method (which handles `model.eval()` and gradient context internally) to ensure correct alignment.

## Testing

Validate that outputs (submission dataframe, local_predictions, local_confidences, global_prediction, global_confidence) are unchanged compared to previous behavior for a sample WAV file.

### CODE REVIEW

File: InferenceSystem/src/model/fastai_inference.py [hunk 1]
```diff
@@ -123,111 +126,131 @@ def predict(self, wav_file_path):
...
+            else:
+                self.model.model.cpu()
+            
+            # Use per-item prediction to ensure correct path-prediction alignment
+            # (get_preds with DataLoader may have different ordering)
+            self.model.model.eval()
+            predictions = []
+            with torch.no_grad():
+                for item in testdb.x:
+                    predictions.append(self.model.predict(item)[2][1])
```
[the remark this answers is not in the document]
Comment (dthaler) [reply InferenceSystem/src/model/fastai_inference.py]: Updated comment:
>             # Per-item prediction loop with torch.no_grad() for memory efficiency
>            # Note: FastAI's predict() handles device placement for inputs internally

**Comment by dthaler** [comment]
Testing with this model showed a significant memory improvement, up to 30%, after running for some time.

**Review by micya — APPROVED** [review]
